controllers/compaction/reconciler.go

```diff
@@ -195,19 +198,27 @@ func (r *Reconciler) reconcileJob(ctx context.Context, logger logr.Logger, etcd
 
 	// Delete job and requeue if the job failed
 	if job.Status.Failed > 0 {
+		if job.Status.StartTime != nil {
+			CompactionJobDurationSeconds.With(prometheus.Labels{druidmetrics.LabelSucceeded: druidmetrics.ValueSucceededFalse}).Set(float64(time.Since(job.Status.StartTime.Time).Seconds()))
```

Comment:
Is there a chance that this metric can be recorded multiple times if the job deletion fails and requeues?

In that case I'd suggest moving this block to after the job has been successfully deleted

Reply:
We actually want it to record a `succeeded: false` metric every time the job fails. That gives us an accurate reading of how many times a job failed. The job can succeed only once, but it can fail multiple times, so the `failed` metric has to captured this many number of times.